types/react-navigation/index.d.ts

```diff
@@ -86,7 +87,7 @@ export type NavigationLeafRoute<Params> = {
    * Params passed to this route when navigating to it,
    * e.g. `{ car_id: 123 }` in a route that displays a car.
    */
-  params?: Params,
+  params: Params,
```

Comment:
I would prefer to type `navigate()` more specifically, say a series of `navigate('routeNameX', params: ParamX)` or so, so that the compiler could do all the checks. The navigation logic implies a screen (local) is mostly coupled with the navigator (global). An ideal type system might be too much complicated to implement currently if possible, I think.